terraform/environments/electronic-monitoring-data/glue-job/parquet_to_csv.py

```diff
@@ -0,0 +1,80 @@
+import sys
+from awsglue.transforms import *
+from awsglue.utils import getResolvedOptions
+from pyspark.context import SparkContext
+from awsglue.context import GlueContext
+from awsglue.job import Job
+
+# Set up Glue context and job
+args = getResolvedOptions(sys.argv, ["JOB_NAME"])
+sc = SparkContext()
+glueContext = GlueContext(sc)
+spark = glueContext.spark_session
+job = Job(glueContext)
+job.init(args["JOB_NAME"], args)
+
+# Source and destination buckets
+source_bucket = "source-bucket-name"
+destination_bucket = "destination-bucket-name"
+source_path = "s3://{}/".format(source_bucket)
+destination_path = "s3://{}/".format(destination_bucket)
+
+# Maximum size for each CSV file (in bytes)
+max_csv_size = 5 * 1024 * 1024 * 1024  # 5GB
+
+
+# Function to write DataFrame to CSV with size check and splitting
+def write_dataframe_to_csv(dataframe, destination_path):
+    # Check the size of the DataFrame
+    dataframe_size = dataframe.rdd.map(lambda x: len(str(x))).reduce(lambda x, y: x + y)
```

Comment:
i'm pretty sure this is giving you how many characters are in the dataframe (converting everything to a string and summing). So the comparison to `max_csv_size` might be a bit wonky